plugins/modules/open_iscsi.py

```diff
@@ -401,7 +403,35 @@ def main():
     if show_nodes:
         result['nodes'] = nodes
 
-    if login is not None:
+    if login is not None and not check_rc:
+        result['devicenodes'] = []
+        for index_target in nodes:
+            loggedon = target_loggedon(module, index_target, portal, port)
+            if (login and loggedon) or (not login and not loggedon):
+                result['changed'] |= False
+                if login:
+                    result['devicenodes'] += target_device_node(index_target)
+            elif not check:
+                if login:
+                    login_result = target_login(module, index_target, check_rc, portal, port)
+                    # give udev some time
+                    time.sleep(1)
+                    result['devicenodes'] += target_device_node(index_target)
+                else:
+                    target_logout(module, index_target)
+                # Check if there are multiple targets on a single portal and
+                # do not mark the task changed if host could not login to one of them
+                if len(nodes) > 1 and login_result == 24:
+                    result['changed'] |= False
+                    result['connection_changed'] = False
+                else:
+                    result['changed'] |= True
+                    result['connection_changed'] = True
+            else:
+                result['changed'] |= True
+                result['connection_changed'] = True
+
+    if login is not None and check_rc:
```

Comment:
Maybe:
```python
    if login is not None:
        if not check_rc:
            result['devicenodes'] = []
            for index_target in nodes:
                ...

        else:  # check_rc
```

Reply:
yep, that seems more compact
I also added some comments to clarify why we use `check_rc`